### PR TITLE
Do not change root-console for non-sle distris

### PR DESCRIPTION
**Pull request by rwx788**
On SLE15 we have gdm running on tty2 and this is desired change. Whereas
on CaaSP there is no such a change, so do not change it when it's CaaSP.

### CODE REVIEW

**Comment by sysrich** [comment]
Why is caasp matching sle_version_at_least('15')?

CaaSP 2.0 is based on SLE 12 SP3 and should be honouring all of the SLE 12 SP3 behaviour

I do not think this is_sle variable should be necessary, but CaaSP should be correctly inheriting sle_version of 12-SP3

**Comment by rwx788** [comment]
@sysrich It breaks https://openqa.suse.de/tests/1125239#step/first_boot/5 on CaaSP. I believe it works there because we won't install gnome. So it will continue work with tty2. Problem is that TW also matches it for whatever reason. https://openqa.opensuse.org/tests/472870

File: lib/susedistribution.pm [hunk 1]
```diff
@@ -264,7 +264,7 @@ sub init_consoles {
         $self->add_console('installation',   'tty-console', {tty => check_var('VIDEOMODE', 'text') ? 1 : 7});
         $self->add_console('install-shell2', 'tty-console', {tty => 9});
         # On SLE15 X is running on tty2 see bsc#1054782
-        $self->add_console('root-console', 'tty-console', {tty => sle_version_at_least('15') ? 6 : 2});
+        $self->add_console('root-console', 'tty-console', {tty => sle_version_at_least('15') && is_sle ? 6 : 2});
```
Comment (okurz) [inline lib/susedistribution.pm]: IMHO sle_version_at_least should internally only check for `is_sle` but we would need to check where it makes problems

Comment (rwx788) [inline lib/susedistribution.pm]: I see the problem, in code like !check_var($version_variable, '12-SP4'). So for versions which are not sle at all it will return true always. I would also ass is_sle to sle_version_at_least, but I wonder if we should make any exception for staging, etc.?

**Comment by sysrich** [comment]
@rwx788 like I said..it's important we get CaaSP honoring the sle_version_at_least

CaaSP 1.0 was mostly 12-SP2 with some 12-SP3, so the behaviour of at_least 12-SP2 would likely have worked

CaasP 2.0 will be 12-SP3 - we need it using the sle_version_at_least 12-SP3 behaviour

CaaSP 4.0 will be 15, which is why I think this is_sle approach is not acceptable - it will just break again in 5 months from now.

**Comment by kravciak** [comment]
Alternative PR - https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3453

We have is_caasp, is_leap, is_tumbleweed, I think is_sle helper function would not hurt (but I did not play with sle15 much, so I might we wrong)

**Comment by okurz** [comment]
TL;DR: use `!is_caasp` to fix it fast, change `sle_version_at_least` later.

It's not that easy. sle_version_at_least returns true by default if it does not match a version, e.g. as in `if_not_sle_then_sle_version_at_least` :-) As SLE by definition based on "Factory first" should always be behind Tumbleweed it makes sense that "by default" the check also returns true. Of course, Leap and CaaSP are both a problem here but just turning around the default will most likely cause many other problems so that should be carefully investigated.

**Comment by kravciak** [comment]
LGTM

**Comment by rwx788** [comment]
I'll create a ticket to approach points mentioned by Richard, as of now, we add this change and I fix openSUSE tests to accept new behavior where it fails.

**Comment by rwx788** [comment]
Ticket to approach points mentioned by Richard: https://progress.opensuse.org/issues/23546
